edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java

```diff
@@ -1160,8 +1162,21 @@ public List getViewableStudentsForUser(Long gradebookId, String userId, List stu
 		List groupIds = new ArrayList(sectionIdStudentIdsMap.keySet());
 		List permsForGroupsAnyCategory = getPermissionsForUserForGroup(gradebookId, userId, groupIds);
 		
-		if (permsForGroupsAnyCategory.isEmpty()) {
-			return viewableStudents;
+		//FIXME: based on our realms checks earlier, sections should already be filtered to one that the
+		//User has access to. May need to make sure they have TA role in those sections
+		if (permsForGroupsAnyCategory.isEmpty()){
+			if(!sections.isEmpty()) {
+				for(Object obj: sections){
+					Permission perm = new Permission();
+					CourseSection section = (CourseSection) obj;
+					perm.setFunction(GraderPermission.GRADE.toString());
```

Comment:
Thanks for catching that. Our TA's always have grade privilege which is probably why that assumption was made. We will change it to check the realm permission for the role.